### PR TITLE
Port the reflection code to TypeInfo.

### PR DESCRIPTION
**Pull request by lbargaoanu**
First step for #404.

### CODE REVIEW

**Comment by lbargaoanu** [comment]
Code analysis complains about unused members (in the extension methods class). It would be easier to let them in and only remove them when the port is over (if we really don't use them).

**Comment by lbargaoanu** [comment]
This touches quite a few files, but the changes are trivial.

**Comment by ploeh** [comment]
Thank you for your interest in contributing to AutoFixture. FTR, I'm currently travelling, and have only limited to time to evaluate pull requests. In due time, I'll work through my backlog, but it may take weeks :flushed: :sweat:

File: Src/AutoFixture/Dsl/NodeComposer.cs [hunk 1]
```diff
@@ -411,7 +412,7 @@ public IPostprocessComposer<T> WithAutoProperties()
             Expression<Func<T, TProperty>> propertyPicker)
         {
             var m = propertyPicker.GetWritableMember().Member;
-            if (m.ReflectedType != typeof(T))
+            if (m.DeclaringType != typeof(T))
```
Comment (ploeh) [inline Src/AutoFixture/Dsl/NodeComposer.cs]: What is the motivation for changing from `ReflectedType` to `DeclaringType`?

Comment (lbargaoanu) [inline Src/AutoFixture/Dsl/NodeComposer.cs]: https://github.com/dotnet/coreclr/issues/926

**Comment by ploeh** [comment]
Thank you for this pull request. As you can see, I had a single question, but apart from that, it's looking good :+1: 

Not only does it build on the CI server, but it also builds on my machine :wink:

File: Src/AutoFixture/Dsl/NodeComposer.cs (hunk 1 above)
Comment (ploeh) [inline Src/AutoFixture/Dsl/NodeComposer.cs]: Thank you; I didn't know that. Apparently, this change isn't an issue, since all tests still pass.

Comment (lbargaoanu) [inline Src/AutoFixture/Dsl/NodeComposer.cs]: I thought so too :) Anyway, I can fix any issues that might appear, but the tests seem to cover a lot.

**Pull request merged**